### DISCUSSION TITLE
/articles/sharepoint-framework-typescript-v5/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#313] [Comments]
While not officially supported, it appears developers can use TypeScript v5.3 in SharePoint Framework projects v1.19 and v1.20. Learn how in this article!
https://www.andrewconnell.com/articles/sharepoint-framework-typescript-v5/

**Comment by sympmarc** [comment]
Really? That’s all it takes? It certainly makes me wonder why the SPFx team wouldn’t simply bring things up to rev in the 1.20 release.

**Reply by andrewconnell** [reply]
![](https://media1.tenor.com/m/IycahZMsX7YAAAAC/seth-meyers-right.gif)

**Comment by stevebeauge** [comment]
I'll never understand why they choose to force to use a specific TS version.

Moreover the build chain first compile TS into JS and then, webpack enters the scene to bundle everything.

Letting us control the TS compilation would have solved a numerous range of toolchain issue.

**Comment by kbeeveer46** [comment]
Don't you also have to change this line in tsconfig.json to 5.3 or no?

`"extends": "./node_modules/@microsoft/rush-stack-compiler-4.7/includes/tsconfig-web.json",`

**Reply by andrewconnell** [reply]
Great catch... I did miss that. Yes, you should update that to point to 5.3. There's no real effect as the **tsconfig-web.json** is identical in the two versions.

I just updated the article (but due to caching, it might take a bit to be reflected).

**Comment by johnnliu** [comment]
This was really helpful to resolve an issue where I was trying to upgrade botframework-webchat to 4.18.0 which had a dependency on typescript 5.0 

took minutes to resolve.
